From c4863ec1933c6f3f9cb922abcc9d366dc542c455 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Wed, 20 Oct 2021 12:08:19 +0200
Subject: [PATCH] net: ethernet: qualcomm: add IPQESS support

Allow compiling the IPQESS driver that supports the
Qualcomm IPQ40xx SoC built-in ethernet controller.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 drivers/net/ethernet/qualcomm/Kconfig  | 8 ++++++++
 drivers/net/ethernet/qualcomm/Makefile | 2 ++
 2 files changed, 10 insertions(+)

--- a/drivers/net/ethernet/qualcomm/Kconfig
+++ b/drivers/net/ethernet/qualcomm/Kconfig
@@ -60,6 +60,14 @@ config QCOM_EMAC
 	  low power, Receive-Side Scaling (RSS), and IEEE 1588-2008
 	  Precision Clock Synchronization Protocol.
 
+config IPQ_ESS
+	tristate "Qualcomm Atheros IPQ ESS support"
+	depends on OF
+	select PHYLINK
+	help
+	  This driver supports the Qualcomm IPQ40xx built-in
+	  ESS ethernet controller.
+
 source "drivers/net/ethernet/qualcomm/rmnet/Kconfig"
 
 endif # NET_VENDOR_QUALCOMM
--- a/drivers/net/ethernet/qualcomm/Makefile
+++ b/drivers/net/ethernet/qualcomm/Makefile
@@ -11,4 +11,6 @@ qcauart-objs := qca_uart.o
 
 obj-y += emac/
 
+obj-$(CONFIG_IPQ_ESS) += ipqess/
+
 obj-$(CONFIG_RMNET) += rmnet/
